src/Button/ButtonBase.tsx

```diff
@@ -76,7 +77,7 @@ const ButtonBase = forwardRef(
             <Box as="span" data-component="buttonContent" sx={getAlignContentSize(alignContent)}>
               {LeadingVisual && (
                 <Box as="span" data-component="leadingVisual" sx={{...iconWrapStyles}}>
-                  <LeadingVisual />
+                  {isValidElementType(LeadingVisual) ? <LeadingVisual /> : LeadingVisual}
```

Comment:
TIL of `isValidElementType`. One odd example I see in [the docs](https://github.com/facebook/react/tree/main/packages/react-is) is that it looks like an element tag name will be `true`:

```ts
ReactIs.isValidElementType("div"); // true
```

If the function actually works like this, it creates some potential weirdness where if the consumer passes a string that happens to match a tag name, we will try to render that tag name instead. But if they pass any other string we will render the string. 

This could be considered expected behavior - `<div />` is valid JSX after all - but it's a pretty strange gotcha. And to be fair I don't expect consumers to pass strings here much since that would make for an odd-looking button, but I think if we support it in the types we should support rendering it.

I wonder if maybe we should check `isValidElementType(LeadingVisual) && typeof LeadingVisual !== "string"`. In which case, maybe we should abstract this logic out to a shared helper util / type.

Reply:
I think that's a good call to force a string to be a string, instead of a maybe element type.   Good call out!